chore(allow-scripts): documentation improvements (#1940)
LavaMoat/LavaMoat: packages/allow-scripts/README.md

> For an overview of LavaMoat tools see [the main README](https://github.com/LavaMoat/LavaMoat/tree/main/README.md)

## Usage

The way this package helps you with your project's secutiry posture is by opting out of running lifecycle scripts from your dependencies by default, while precisely allowing the ones you actually need.

1. Run `allow-scripts setup` **once** per project to opt out of running scripts first
2. Run `allow-scripts auto` to populate or update the allowlist in package.json
3. Run `allow-scripts run` after every install to run the lifecycle scripts

We recommend using the globally installed `@lavamoat/allow-scripts` for initial setup and permanently installing it in the project for further use so that every contributor has the ability to run the allowed scripts.

> 

Reviewer: ```suggestion
## Usage

`@lavamoat/allow-scripts` increases your project's security posture by first _denying_ all lifecycle scripts from running, then _allowing_ the ones you actually need.  The workflow looks like this:

1. Run `allow-scripts setup` **once** per project to opt-out of running scripts
2. Run `allow-scripts auto` to populate or update the allowlist (configured in `package.json`)
3. Run `allow-scripts run` after every install to run the lifecycle scripts

We recommend using the globally installed `@lavamoat/allow-scripts` for initial the initial `setup`. Thereafter, install it as a project-local development dependency to enable contributors to execute allowed scripts.
```

### Install globally

```sh
npm i -g @lavamoat/allow-scripts
```

> Be sure to include the `@lavamoat/` namespace in the package name.

Reviewer: ```suggestion
**Be sure to include the `@lavamoat/` namespace in the package name.**
```

allow-scripts setup
```

Adds (or updates) a `.yarnrc` or `.npmrc` (depending on which lockfile is present) file in the current package and appends `ignore-scripts=true`.

`pnpm` ignores scripts by default, but the `.npmrc` file is generated by default. It's worth keeping in case someone goes for `npm install` before they notice pnpm is used in the repo.

Immediately after creating the configuration file, allow-scripts setup adds the devDependency `@lavamoat/preinstall-always-fail`. It is intended to remain disallowed. Adding this package to a project **mitigates** against accidentally running any lifecycle scripts by throwing an error during the `preinstall` script execution in case someone deletes the `.npmrc`

### Populate the allowlist

> Automatically generates and writes a configuration into `package.json`, setting new policies as `false` by default. Set the packages you need to run lifecycle scripts to true.

Reviewer: ```suggestion
`allow-scripts auto` generates and writes a configuration into your `package.json`, setting new policies to `false` (deny) by default. 

After execution, you may enable packages' lifecycle scripts by setting the associated policies to `true` (allow).
```

If allowScripts has an item set to true, the lifecycle scripts of that package will be executed in the run phase if the current installed version is still matching.

> **Note** allow-scripts is pinning versions of the dependencies with lifecycle since v5. Items set to false will get automatically updated to the current version detected. To avoid churn in the list, you can remove the `#version` part from a _disallowed_ item and it will be denied regardless of version. Versions are mandatory for allowed packages in case of maintainer compromise. You can opt out of versions with the `--skip-versions` flag on both `auto` and `run` commands.

> > **Tip** While you can configure all install scripts that you've been running to date as allowed, it's best to limit the number of them in case a package with pre-existing install script gets exploited. To figure out which packages' scripts can be ignored, try [can-i-ignore-scripts](https://www.npmjs.com/package/can-i-ignore-scripts)

Reviewer: use `[!TIP]`
